### ISSUE TITLE
When using accelerate with deepspeed and peft, any kwargs passed to Accelerate( ) are not used?

### ISSUE THREAD

**Issue by sandeepchittilla** [huggingface/accelerate#1331]
**A bit of context**: 

This issue was originally posted [here](https://github.com/lvwerra/trl/issues/304)

I am launching my script with `accelerate launch my_script.py` with a custom accelerate config file. I'm running this in a distributed training setting across 4 GPUs with deepspeed and gradient accumulation.

I save the model after every step, and after a few steps + saving, I suddenly get NCCL Timeout errors (default is 30mins). So, I increase the timeout to 1.5hr by passing `kwargs_handlers=[InitProcessGroupKwargs(timeout=timedelta(seconds=5_400))]` during Accelerator initialization.

However, these values are not used because the accelerator has already been initialized probably since I launch it via cli?

Is there a workaround for this? Or an option to edit the state in case some previously unset parameter is passed? Or .... I'm missing something?

**The error**:

![image](https://user-images.githubusercontent.com/62606281/232790900-0d377a54-d29e-40f8-a056-9d507002107c.png)

**How the model is saved**:

```python
if trainer.accelerator.is_main_process:
        if trainer.is_distributed:
            torch.distributed.barrier()
        os.makedirs(os.path.join(output_dir, f"checkpoint_{chkpt_name}"), exist_ok=True)
        trainer._save_pretrained(os.path.join(output_dir, f"checkpoint_{chkpt_name}"))

...
# this is a mock class to show how the _save_pretrained( ) is implemented

from huggingface_hub import PyTorchModelHubMixin
class Trainer(PyTorchModelHubMixin):
     ...
    def _save_pretrained(self, save_directory: str) -> None:
        self.accelerator.unwrap_model(self.model).save_pretrained(save_directory)
        self.tokenizer.save_pretrained(save_directory)
      
```

**Comment by sgugger** [comment]
cc @pacman100

**Comment by sandeepchittilla** [comment]
fyi, using `accelerate 0.16.0` and `torch 2.0.0` and `deepspeed 0.8.0`

**Comment by pacman100** [comment]
Hello, could you try the latest accelerate and deepspeed versions and let us know if the issue still persist?
